### ISSUE TITLE
Give better error messages when inheriting from actor

### ISSUE THREAD

**Issue by pcmoritz** [ray-project/ray#5518]
If we do the following:

```
In [12]: @ray.remote 
    ...: class A(object): 
    ...:     def __init__(self): 
    ...:         pass 
    ...:                                                                                                                                                                             

In [13]: class B(A): 
    ...:     def __init__(self): 
    ...:         pass
```

We get the following error:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-13-d1120a93dc7b> in <module>
----> 1 class B(A):
      2     def __init__(self):
      3         pass
      4 

TypeError: __init__() missing 5 required positional arguments: 'num_cpus', 'num_gpus', 'memory', 'object_store_memory', and 'resources'
```

It should instead say something like "cannot inherit from actor".

**Comment by pcmoritz** [comment]
something like this might work: http://code.activestate.com/recipes/141933-forbidding-inheritance/

**Comment by edoakes** [comment]
Looks like fixing this will require a workaround or solving a bit of a deeper problem. The cause of the current error message is that our `@ray.remote` decorator for classes is not returning the correct type. It should be a a `type` metaclass but instead is returning an ordinary `object` class. This is why we are getting the above error message - Python calls `__init__` on what it thinks is a `type` metaclass but in reality it is just our `ActorClass`:

```
import ray
ray.init()

class RealClass:
    pass

@ray.remote
class Base:
    pass

print(type(RealClass))   # <class 'type'>
print(type(RealClass())) # <class '__main__.RealClass'>
print(type(Base))        # should be <class 'type'> but <class ray.actor.ActorClass>'
```

To fix this, we need to use the `type(<name>, <base_classes>, <attribute_dict>)` constructor instead of the `ActorClass` constructor, but doing so naiively breaks the API because the resulting `type` metaclass doesn't have the expected `.remote()` method.

Will try to fix this in the near future.

cc @dhroth @kiddyboots216

**Comment by krfricke** [comment]
Closed by #5598
